OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -160,7 +160,7 @@ public override bool Tick(Actor self)
 
 			var firstFacing = self.World.Map.FacingBetween(mobile.FromCell, nextCell.Value.Cell, mobile.Facing);
 
-			if (mobile.Info.CanMoveBackward && self.World.WorldTick - startTicks < mobile.Info.BackwardDuration && Math.Abs(firstFacing.Angle - mobile.Facing.Angle) > 256)
+			if (mobile.Info.CanMoveBackward && (mobile.Info.MaxBackwardCells < 0 || path.Count < mobile.Info.MaxBackwardCells) && (mobile.Info.BackwardDuration < 0 || self.World.WorldTick - startTicks < mobile.Info.BackwardDuration) && Math.Abs(firstFacing.Angle - mobile.Facing.Angle) > 256)
```

Comment:
```suggestion
			if (mobile.Info.CanMoveBackward
				&& (mobile.Info.MaxBackwardCells < 0 || path.Count < mobile.Info.MaxBackwardCells)
				&& (mobile.Info.BackwardDuration < 0 || self.World.WorldTick - startTicks < mobile.Info.BackwardDuration)
				&& Math.Abs(firstFacing.Angle - mobile.Facing.Angle) > 256)
```

Please split it up so it is readable

Reply:
fixed